src/UNL/MediaHub/Media/Image.php

```diff
@@ -1,15 +1,85 @@
 <?php
-class UNL_MediaHub_Media_Image extends UNL_MediaHub_Media
+class UNL_MediaHub_Media_Image
 {
-    /**
-     * Get by ID
-     *
-     * @param int $id The id of the feed to get
-     *
-     * @return UNL_MediaHub_Media
-     */
-    static function getById($id)
+    protected $options;
+    
+    public function __construct(array $options)
     {
-        return Doctrine::getTable(__CLASS__)->find($id);
+        $this->options = $options;
+
+        //Validate input
+        if (!isset($this->options['id'])) {
+            throw new \Exception('You must provide a media_id', 400);
+        }
     }
-}
\ No newline at end of file
+    
+    public function getThumbnail()
```

Comment:
It would be better if this method returned a file system path instead of the file data. That way you can send the file without loading it into memory.